ocs_ci/ocs/fiojob.py

```diff
@@ -0,0 +1,233 @@
+    ceph_total_stored = 0
+    for pool in ceph_df_dict["pools"]:
+        ceph_total_stored += pool["stats"]["stored"]
+        if pool["name"] == ceph_pool_name:
+            ceph_pool = pool
+    if ceph_pool is None:
+        logger.error((
+            f"pool {ceph_pool_name} was not found "
+            f"in output of `ceph df`: {ceph_df_dict}"))
+    # If the following assert fail, the problem is either:
+    #  - name of the pool has changed (when this happens before GA, it's
+    #    likely ocs-ci bug, after the release it's a product bug),
+    #  - pool is missing (likely a product bug)
+    # either way, the fixture can't continue ...
+    assert ceph_pool is not None, f"pool {ceph_pool_name} should exist"
+    # ... to compute PVC size (values in bytes)
+    total = ceph_pool["stats"]["max_avail"] + ceph_total_stored
+    max_avail_gi = ceph_pool['stats']['max_avail'] / 2**30
+    logger.info(f"MAX AVAIL of {ceph_pool_name} is {max_avail_gi} Gi")
+    target = total * target_percentage
+    to_utilize = target - ceph_total_stored
+    pvc_size = round(to_utilize / 2**30)  # GiB
+    logger.info(
+        f"to reach {target/2**30} Gi of total cluster utilization, "
+        f"which is {target_percentage*100}% of the total capacity, "
+        f"utilization job should request and fill {pvc_size} Gi volume")
+    return pvc_size
+
+
+def fio_to_dict(fio_output):
+    """"
+    Parse fio output and provide parsed dict it as a result.
+    """
+    fio_output_lines = fio_output.splitlines()
+    line_num = 0
+    for line_num, line in enumerate(fio_output_lines):
+        if line == "{":
+            break
+        else:
+            logger.info(line)
+    fio_parseable_output = "\n".join(fio_output_lines[line_num:])
+    try:
+        fio_report = yaml.safe_load(fio_parseable_output)
+    except yaml.parser.ParserError as ex:
+        logger.error("json output from fio can't be parsed: %s", ex)
+        raise ex
+    return fio_report
+
+
+def get_timeout(fio_min_mbps, pvc_size):
+    """
+    Compute how long we will let the job running while writing data to the
+    volume.
+
+    Args:
+      fio_min_mbps (int): minimal write speed in MiB/s
+      pvc_size (int): size of PVC in GiB, which will be used to writing
+
+    Returns: write_timeout in seconds
+    """
+    # based on min. fio write speed of the enviroment ...
+    logger.info(
+        "Assuming %.2f MB/s is a minimal write speed of fio.", fio_min_mbps)
+    # ... we compute max. time we are going to wait for fio to write all data
+    min_time_to_write_gb = 1 / (fio_min_mbps / 2**10)
+    write_timeout = pvc_size * min_time_to_write_gb  # seconds
+    logger.info((
+        f"fixture will wait {write_timeout} seconds for the Job "
+        f"to write {pvc_size} Gi data on OCS backed volume"))
+    return write_timeout
+
+
+def wait_for_job_completion(namespace, timeout, error_msg):
+    """
+    This is a WORKAROUND of particular ocsci design choices: I just wait
```

Comment:
If this is workaround can you open issue for this workaround and link it here?

Reply:
This is not [a product workaround in a sense as described in ocs-ci documentation](https://ocs-ci.readthedocs.io/en/latest/docs/workarounds.html), but a workaround to overall design choices done in ocs-ci.